### ISSUE TITLE
Add Method API endpoints

### ISSUE THREAD

**Issue by JannKleen** [coordnet/coordnet#273]
Depends on #269.

Method endpoint:
 - Filters:
   - Own methods, methods explicitly granted access to, public methods (@betamax does that sound like what you'd need, before we decide on the profile features?)
   - Setting / changing / removing permissions
   - Editing title, description, authors and image
   - Author validation
   - (optional) check image size / resolution / format (or create new issue for it)
